periodictable/nsf.py

```diff
@@ -466,32 +472,57 @@ def scattering(self, wavelength=ABSORPTION_WAVELENGTH):
         if not self.has_sld():
             return None, None, None
 
-        b_c = self.b_c
-        sigma_s = self.total
-        sigma_a = self.absorption*wavelength/ABSORPTION_WAVELENGTH
-
         number_density = self._number_density*1e-24
+        b_c, b_pp, sigma_s = self.xs_by_wavelength(wavelength)
+        return _calculate_scattering(number_density, wavelength, b_c, b_pp, sigma_s)
 
-        # PAK 2017-04-21: compute incoherent xs from total xs
-        sigma_c = 4*pi/100 * b_c**2
-        sigma_i = max(sigma_s - sigma_c, 0.)
+def _calculate_scattering(number_density, wavelength, b_c, b_pp, sigma_s):
+    #print("in scat", number_density, wavelength, b_c, b_pp, sigma_s)
+    # Compute SLD
+    sld_re = 10*number_density * b_c
+    sld_im = -10*number_density * b_pp
 
-        # Compute SLD
-        sld_re = number_density * b_c * 10
-        sld_im = number_density * sigma_a / (2 * wavelength) * 0.01
-        sld_inc = number_density * sqrt(sigma_i / (4*pi/100)) * 10
+    # PAK 2017-04-21: compute incoherent xs from total xs
+    sigma_c = 4*pi/100 * b_c**2
+    sigma_i = np.maximum(sigma_s - sigma_c, 0.)
+    b_i = sqrt(sigma_i / (4*pi/100))
+    sld_inc = number_density * b_i * 10
+
+    # Compute absorption cross section
+    sigma_a = -2000 * b_pp * wavelength
```

Comment:
Also, where did the factors of 100 and 2000 come from in this section?  It might be described somewhere in the function docs but it would help a lot to describe it here as well.